### ISSUE TITLE
Let me tell you what was in the source code

### ISSUE THREAD

**Issue by X3eRo0** [xrmon/ctf-writeups#1]
What I planned was that I will store encrypted flag in the binary and I will use a cipher to decrypt it and check with user input.
I missed out on some of the very essential part of crackmes that you don't compare keys like that. This was mainly because I just started making these and before that I used to solve them. But what I liked in this write-up was that if you were pretty close to the intended solution.

I used vigenere cipher to encrypt the flag and use the key as one of the strings in the binary itself.
To make it a bit more complicated, I decided to extract all the strings and several them in a file called "dontlook" and then a grep like function which takes a small string as an argument and returns a string with matches with the argument, just like grep.

Now the. Question rises which string should I use to encrypt the flag.
There was a restriction that I should not use key with any non-alphanumeric charecters.
I could use that but it all depends on YOUR VIGENERE encrypt/decrypt program.
Since most of the online tool don't support key with non-alphanumeric charecters I decided to go with normal strings.
There were not a lot of them except some the strings which were already embedded and used somewhere else
Like the string "dontlook" which was used to open the file "dontlook". In the end I decided to go with that and as it turns out that was a bad idea because some people just guessed that the flag might be encrypted with vigenere cipher and did some cool things to get the flag without actually reversing anything. 
The whole idea was 
[1] Open the binary itself
[2] extract all strings and save them in a file
[3] grep for a perticular string which would be key
[4] use that string to decrypt flag and use it to validate user input.

I've learned from my mistakes and hopefully next time these crackmes will actually be worth the time you guys spend solving.

**Comment by xrmon** [comment]
That's very interesting, thanks for getting back to me!

I did get a way into the static reversing, and I found the function which searched for strings containing "don". I had no idea it was Vigenere however, as shortly after I noticed the checking function and decided to have a go dynamically.

Out of curiosity, there were a couple of things in the challenge that I hadn't seen before: The main function seemed to be at a random location each time (like ASLR, but I've never seen that on the .text section before), and all global variables were referenced relative to EBX, meaning radare couldn't recover them. Did you do these things intentionally or was your compiler just doing weird things?

Thanks for the challenge, I had good fun with the CTF and it was a great chance to practise my dynamic reversing skills.
